engine/src/main/java/org/hibernate/search/backend/impl/lucene/works/DeleteByQueryWorkDelegate.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend.impl.lucene.works;
+
+import org.apache.lucene.index.IndexWriter;
+import org.apache.lucene.index.Term;
+import org.apache.lucene.search.BooleanClause;
+import org.apache.lucene.search.BooleanQuery;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.TermQuery;
+import org.hibernate.search.backend.DeleteByQueryLuceneWork;
+import org.hibernate.search.backend.DeleteByQuerySupport;
+import org.hibernate.search.backend.IndexingMonitor;
+import org.hibernate.search.backend.LuceneWork;
+import org.hibernate.search.backend.DeletionQuery;
+import org.hibernate.search.engine.ProjectionConstants;
+import org.hibernate.search.exception.SearchException;
+import org.hibernate.search.store.Workspace;
+import org.hibernate.search.util.impl.ScopedAnalyzer;
+import org.hibernate.search.util.logging.impl.Log;
+import org.hibernate.search.util.logging.impl.LoggerFactory;
+
+/**
+ * Stateless implementation that performs a <code>DeleteByQueryWork</code>.
+ *
+ * @author Martin Braun
+ * @see LuceneWorkVisitor
+ * @see LuceneWorkDelegate
+ */
+class DeleteByQueryWorkDelegate implements LuceneWorkDelegate {
+
+	private static final Log log = LoggerFactory.make();
+	protected final Workspace workspace;
+
+	DeleteByQueryWorkDelegate(Workspace workspace) {
+		this.workspace = workspace;
+	}
+
+	@Override
+	public void performWork(LuceneWork work, IndexWriter writer, IndexingMonitor monitor) {
+		DeleteByQueryLuceneWork deleteWork = (DeleteByQueryLuceneWork) work;
+
+		final Class<?> entityType = work.getEntityClass();
+		final DeletionQuery query = deleteWork.getDeletionQuery();
+
+		log.tracef( "Removing all %s matching Query: %s", entityType.toString(), query.toString() );
+
+		BooleanQuery entityDeletionQuery = new BooleanQuery();
+
+		// TODO: we could let experts supply their own means to convert back to
```

Comment:
I'd drop this all :)
If there's need to make things pluggable, it should not be too hard to figure out.